discord/application.go

```diff
@@ -97,6 +97,43 @@ type CommandOption struct {
 	Required    bool                  `json:"required"`
 	Choices     []CommandOptionChoice `json:"choices,omitempty"`
 	Options     []CommandOption       `json:"options,omitempty"`
+
+	// If this option is a channel type, the channels shown will be restricted to these types
+	ChannelTypes []ChannelType `json:"-"`
+}
+
+func (c CommandOption) MarshalJSON() ([]byte, error) {
+	type RawOption CommandOption
+	option := struct {
+		RawOption
+		ChannelTypes []uint16 `json:"channel_types,omitempty"`
+	}{RawOption: RawOption(c)}
+
+	// []uint8 is marshalled as a base64 string, so we marshal a []uint64 instead.
+	if len(c.ChannelTypes) > 0 {
+		for _, t := range c.ChannelTypes {
```

Comment:
It's better to preallocate the `option.ChannelTypes` slice here:

```suggestion
		option.ChannelTypes = make([]uint16, 0, len(c.ChannelTypes))
		for _, t := range c.ChannelTypes {
```